cider/exec/nextgen/operators/Column2RowNode.cpp

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "exec/nextgen/operators/Column2RowNode.h"
+
+#include "exec/module/batch/ArrowABI.h"
+#include "exec/nextgen/jitlib/base/ValueTypes.h"
+#include "exec/nextgen/operators/expr.h"
+
+namespace cider::exec::nextgen::operators {
+void Column2RowTranslator::consume(Context& context) {
+  codegen(context);
+}
+
+void Column2RowTranslator::codegen(Context& context) {
+  auto func = context.query_func_;
+  auto inputs = node_.getExprs();
+  // for row loop
+  auto arrow_pointer = func->getArgument(0);
+  auto index = func->createVariable(JITTypeTag::INT64, "index");
+  index = func->createConstant(JITTypeTag::INT64, 0l);
+  auto len = func->createVariable(JITTypeTag::INT64, "len");
+  // len means rows length
+  len = func->emitRuntimeFunctionCall(
+      "extract_arrow_array_len",
+      JITFunctionEmitDescriptor{.ret_type = JITTypeTag::INT64,
+                                .params_vector = {arrow_pointer.get()}});
+
+  func->createLoopBuilder()
+      ->condition([&index, &len]() { return index < len; })
+      ->loop([&]() {
+        col2RowConvert(inputs, func, index);
+        CHECK(successor_);
```

Comment:
Needn't ```CHECK``` here, the validation of pipeline will be guaranteed by Parser & Transformer.

Reply:
Have deleted this **CHECK** under operators